graphql-protocol/src/main/java/com/linecorp/armeria/server/graphql/protocol/AbstractGraphqlService.java

```diff
@@ -166,7 +226,15 @@ private static Map<String, Object> toMap(@Nullable String value) throws JsonProc
         if (Strings.isNullOrEmpty(value)) {
             return ImmutableMap.of();
         }
-        return parseJsonString(value);
+        return parseJsonString(value, JSON_MAP);
+    }
+
+    private static String getValueFromMultipartParam(String name, ListMultimap<String, String> params) {
+        final List<String> list = params.get(name);
+        if (list.isEmpty() || Strings.isNullOrEmpty(list.get(0))) {
+            throw new IllegalArgumentException(name + " form field is missing");
+        }
```

Comment:
What do you think of also strictly validating that there is only a single `operations` and `map` parameter to prevent user confusion?

```
if (list.size() > 1) {
    throw new IllegalArgumentException("More than one 'operation' received.");
}
```

Reply:
good idea. Thanks for the suggestion. 👍